src/game.cpp

```diff
@@ -10584,14 +10592,15 @@ bool game::handle_liquid( item &liquid, item * const source, const int radius,
     }
 
     menu.query();
+    refresh_all();
     const size_t chosen = static_cast<size_t>( menu.ret );
     if( chosen >= actions.size() ) {
         add_msg( _( "Never mind." ) );
         // Explicitly canceled all options (container, drink, pour).
         return false;
     }
-    actions[chosen]();
-    return true;
+
+    return actions[chosen]();
```

Comment:
This kind of changes the meaning of the return value, please make sure it still fits the documented eaning (see game.h).

Returning true means the player wanted to handle the liquid, but may have failed to actually do anything with it (e.g. choose to pour it into a container, but could not find a suitable one, or choose to pour into a keg on the ground, but that was already full). It should return `false` when the player explicitly canceled.